geo/src/algorithm/bool_ops/op.rs

```diff
@@ -173,15 +209,22 @@ impl<T: Float, S: Spec<T>> Proc<T, S> {
     }
 }
 
+/// An edge of a shape.
 #[derive(Clone)]
 struct Edge<T: Float, S: Spec<T>> {
+    /// The geometry of the edge.
     geom: LineOrPoint<T>,
+    /// The index of the shape this edge belongs to.
     idx: usize,
+    /// The region of this edge when the piece is going in the same direction as
```

Comment:
I'm pretty sure this is wrong. I can sorta understand the intent, but it doesn't make sense to me exactly what the check below means.

Reply:
@rmanoka Can you shed any light on this?